WebService/Startup.cs

```diff
@@ -109,6 +109,12 @@ private void StartAgents()
 
             this.simulationAgent = this.ApplicationContainer.Resolve<ISimulationAgent>();
             this.simulationAgent.StartAsync();
+
+            var config = this.ApplicationContainer.Resolve<IConfig>();
+            if (!string.IsNullOrEmpty(config.SeedTemplate))
```

Comment:
Seeding should happen only once, only just after the deployment, only in 1 VM. This approach will run the Seed call in every VM and every time the service starts (e.g. after a crash), leading to unexpected situations.

Reply:
Fixed by adding a flag in storage and checking for that before seeding